examples/flutter_gallery/lib/demo/customized/run_stats_content.dart

```diff
@@ -0,0 +1,123 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'animation_helpers.dart';
+import 'stats/full_stats.dart';
+import 'stats/my_stats_bar.dart';
+import 'stats/run_summary.dart';
+
+class RunStatsContent extends StatefulWidget {
+  const RunStatsContent({ Key key, @required this.onHeaderTapped, this.peekAmount = 70.0 })
+      : assert(onHeaderTapped != null),
+        super(key: key);
+
+  final VoidCallback onHeaderTapped;
+
+  /// The stats content will peek out this amount. This will also be the
+  /// size of the "show details" bar.
+  final double peekAmount;
+
+  @override
+  RunStatsContentState createState() => RunStatsContentState();
+}
+
+class RunStatsContentState extends State<RunStatsContent> {
+
+  final GlobalKey<MyStatsBarState> _myStatsBarKey = GlobalKey<MyStatsBarState>();
+  Widget _myStatsBar;
+  final GlobalKey<RunSummaryState> _runSummaryKey = GlobalKey<RunSummaryState>();
+  Widget _runSummary;
+  final GlobalKey<FullStatsState> _fullDetailsKey = GlobalKey<FullStatsState>();
+  Widget _fullDetails;
+
+  AnimationController _animationController;
+
+  Animation<double> _rotationAnimation;
+  Animation<double> _runnerFadeAnimation;
+  Animation<double> _pathFadeAnimation;
+  Animation<double> _statsSummaryAnimation;
+  Animation<double> _numberCounterAnimation;
+
+  @override
+  Widget build(BuildContext context) {
+    _myStatsBar ??= MyStatsBar(
+      key: _myStatsBarKey,
+      onTapped: widget.onHeaderTapped,
+      height: widget.peekAmount,
+    );
+    _runSummary ??= RunSummary(key: _runSummaryKey);
+    _fullDetails ??= FullStats(key: _fullDetailsKey);
+    final MediaQueryData mediaQueryData = MediaQuery.of(context);
+    return Container(
+      height: mediaQueryData.size.height - mediaQueryData.padding.top,
```

Comment:
Consider using SafeArea in general. Also make sure to test the iPhone X on all orientations.

Reply:
There are particular reasons why we don't use them in this case. I definitely would use them otherwise.